src/rules/prefer-nullish-coalescing-boolean-props.ts

```
@@ -0,0 +1,552 @@
+import { createRule } from '../utils/createRule';
+
+type MessageIds = 'preferNullishCoalescing';
+
+/**
+ * Determines if a node is within a boolean context in JSX props or other boolean contexts
+ */
+function isInBooleanContext(node: TSESTree.Node): boolean {
+  let current: TSESTree.Node | undefined = node;
+
+  // Traverse up the AST to find if we're in a boolean context
+  while (current && current.parent) {
```

Biome (1.9.4)

[error] Change to an optional chain.

Unsafe fix: Change to an optional chain.


(lint/complexity/useOptionalChain)